virtcontainers/ich.go

```diff
@@ -0,0 +1,813 @@
+	// to fetch if this is the first time the hypervisor is created.
+	if err := ich.store.Load(store.Hypervisor, &ich.info); err != nil {
+		ich.Logger().WithField("function", "createSandbox").WithError(err).Info("No info could be fetched")
+	}
+	
+	ichPath, err := ich.ichPath()
+	if err != nil {
+		return err
+	}
+	
+	// Add memory to the cloud hypervisor
+	memory, err := ich.arch.appendMemory(ich.config.MemorySize, "/dev/shm")
+	if err != nil {
+		return err
+	}
+
+	// Add vcpu to the cloud hypervisor
+	vcpu, err := ich.arch.appendProcessors(ich.config.NumVCPUs)
+	if err != nil {
+		return err
+	}
+	
+	// Add the kernel parameters and kernel path
+	kernelPath, err := ich.config.KernelAssetPath()
+	if err != nil {
+		return err
+	}
+	kernel := govmmIch.Kernel{
+		Path:       kernelPath,
+		Params:     ich.kernelParameters(),
+	}
+ 	 
+	ichConfig := govmmIch.Config{
+		Name:        fmt.Sprintf("sandbox-%s", ich.id),
+		UUID:        ich.state.UUID,
+		Path:        ichPath,
+		PidFile:     filepath.Join(store.RunVMStoragePath(), ich.id, "pid"),
+		Memory:		 memory,
+		VCPU:		 vcpu,
+		Kernel:      kernel,
+	}
+	
+	// Add RNG device to hypervisor
+	rngDev := config.RNGDev{
+		ID:       rngID,
+		Filename: ich.config.EntropySource,
+	}
+	ichConfig.Devices, err = ich.arch.appendRNGDevice(ichConfig.Devices, rngDev)
+	
+	// Add the hybrid vsock device to hypervisor
+	vsockDev := types.HybridVSock {
+		UdsPath: 	ich.socketPath,
+		Port: 		1024,
+	}
+	ichConfig.Devices, err = ich.arch.appendHybridVSock(ichConfig.Devices, vsockDev)
+	if err != nil {
+		return err
+	}
+	
+	// Add the root disk to the hypervisor
+	imagePath, err := ich.config.ImageAssetPath()
+	if err != nil {
+		return err
+	}
+
+	if imagePath != "" {
+		ichConfig.Devices, err = ich.arch.appendImage(ichConfig.Devices, imagePath)
+		if err != nil {
+			return err
+		}
+	}
+	 
+	// Add the virtio-fs to the hypervisor
+	vfsdSockPath, err := ich.vfsdFSSocketPath(ich.id)
+	if err != nil {
+		return err
+	}	
+	if vfsdSockPath != "" {
+		ichConfig.Devices, err = ich.arch.appendVirtualFilesystem(ichConfig.Devices, "kataShared", vfsdSockPath)
+		if err != nil {
+			return err
+		}
+	}
+	
+	// Add the serial console to the cloud hypervisor
+	if ich.config.Debug {
+		ichConfig.Devices, err = ich.arch.appendSerialConsole(ichConfig.Devices, "file=/tmp/ich.log")
+	} else {
+		ichConfig.Devices, err = ich.arch.appendSerialConsole(ichConfig.Devices, "off")
+	}
+	if err != nil {
+		return err
+	}
+	
+	// Add the virtio console to the cloud hypervisor
+	ichConfig.Devices, err = ich.arch.appendVirtioConsole(ichConfig.Devices, "off")
+	if err != nil {
+		return err
+	}
+	 
+	// Add the http api endpoint to the cloud hypervisor by default
+//	apiSocketPath, err := ich.apiSocketPath(ich.id)
+//	if err != nil {
+//		return err
+//	}
+	
+//	ichConfig.Devices, err = ich.arch.appendApiSocket(ichConfig.Devices, apiSocketPath)
+//	if err != nil {
+//		return err
+//	}
```

Comment:
should remove commented out code

Reply:
done. moved to rev 2 of the driver to be done next week